File: 30days/10/10.rb
#!/usr/bin/env ruby

n = gets.strip.to_i

binary = n.to_s(2)

ones = 0
target = 0

(0...binary.length).each do |i|
    if binary[i] == "1"
        ones += 1
        target = ones if ones > target
    else
        ones = 0
    end
end

puts target

File: 30days/10/10.py
#!/usr/bin/env python

n = int(input().strip())

binary = bin(n)[2:]

ones = 0
target = 0

for i in range(0, len(binary)):
    if binary[i] == "1":
        ones += 1
        if ones > target:
            target = ones
    else:
        ones = 0

print(target)
